SU-SWS/decanter
# 508: Create media component

# READY FOR REVIEW

# Summary
- Create new Media component that accommodates images, videos, audio, embeds (including YouTube videos for example). Will be used in more complex components such as Cards. 
- Captions are included as an optional element in this component.
- The `@aspect-ratio` mixin can be applied to this component to create a fluid container with fixed aspect ratio, with the media content filling the container (except audio which doesn't apply).
- The default `.su-media` takes an open HTML field for the media content, which includes things like `<iframe>` from YouTube embed.
- The aspect ratio mixin PR #445 was merged into this branch so I can test this component to work with the mixin.

# Needed By (Date)
- Whenever

# Urgency
- Not Urgent

# Steps to Test

1. Pull this branch and compile styleguide.
2. Look at Components -> Media and check that the different variants (image, video, audio embedded video) work as expected.
3. In the components/_media.scss file, ddd `@include aspect-ratio(1,1);` to the `.su-media__wrapper` element, do `npm run build` again.
4. Refresh and check that the default `.su-media` and every variant except the audio one has the media element filling a square container.
5. Change data in media.json file, do `npm run styleguide` and see that it works as expected.
6. Check that the .su-hero--caption variant looks good with the tweaked caption styles.

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- #508 
- #490
- #445 
- #451 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
